src/ViewTrait.php

```diff
@@ -251,6 +250,14 @@ public function setTheme(?Theme $theme): static
         return $this;
     }
 
+    public function withTheme(?Theme $theme): static
+    {
+        $new = clone $this;
+        $new->state = clone $this->state;
```

Comment:
State do not should clone, parameteres and blocks from state should keep. 

I think need extract theme to separate `ThemeState`  by analogy with `LocaleState`.

Reply:
> State should not be cloned, parameters and blocks from the state should be kept.
> 
> I think need extract theme to separate `ThemeState` by analogy with `LocaleState`.

Done